### PR TITLE
Add Trino 429 release notes

### LINKED ISSUE #19270: Release notes for 429

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 429 release. Fixes #19270  

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 4 Oct 2023

* #19238 ✅ rn ✅ docs
* #19173 ✅ rn ✅ docs
* #19269 ✅ rn ✅ docs
* #19265 ✅ rn ✅ docs
* #19160 ✅ rn ✅ docs

## 5 Oct 2023

* #19149 ✅ rn ✅ docs
* #19204 ✅ rn ✅ docs
* #19243 ✅ rn ✅ docs
* #19268 ✅ rn ✅ docs
* #19279 ✅ rn ✅ docs
* #19280 ✅ rn ✅ docs
* #19283 ✅ rn ✅ docs
* #19248 ✅ rn ✅ docs
* #19252 ✅ rn ✅ docs
* #19261 ✅ rn ✅ docs
* #19276 ✅ rn ✅ docs

## 6 Oct 2023

* #19291 ✅ rn ✅ docs
* #19218 ✅ rn ✅ docs
* #19274 ✅ rn ✅ docs
* #19293 ✅ rn ✅ docs
* #19297 ✅ rn ✅ docs
* #19299 ✅ rn ✅ docs
* #19249 ✅ rn ✅ docs
* #19301 ✅ rn ✅ docs
* #19295 ✅ rn ✅ docs
* #19289 ✅ rn ✅ docs

## 7 Oct 2023

* #19304 ✅ rn ✅ docs

## 8 Oct 2023

* #19259 ❌ rn ✅ docs
* #19300 ✅ rn ✅ docs
* #19309 ✅ rn ✅ docs

## 9 Oct 2023

* #19310 ✅ rn ✅ docs
* #19311 ✅ rn ✅ docs
* #19312 ✅ rn ✅ docs
* #18981 ✅ rn ✅ docs
* #19315 ✅ rn ✅ docs
* #19323 ✅ rn ✅ docs
* #19325 ✅ rn ✅ docs

## 10 Oct 2023

* #13981 ✅ rn ✅ docs
* #19328 ✅ rn ✅ docs
* #19317 ✅ rn ✅ docs
* #19321 ✅ rn ✅ docs
* #19329 ✅ rn ✅ docs
* #19330 ✅ rn ✅ docs
* #19327 ✅ rn ✅ docs
* #19296 ✅ rn ✅ docs
* #19314 ✅ rn ✅ docs
* #19324 ✅ rn ✅ docs
* #19342 ✅ rn ✅ docs

## 11 Oct 2023

* #18753 ✅ rn ✅ docs
* #19201 ✅ rn ✅ docs
* #19319 ✅ rn ✅ docs
* #19332 ✅ rn ✅ docs
* #19331 ✅ rn ✅ docs
* #19224 ✅ rn ✅ docs
* #19344 ✅ rn ✅ docs
* #19334 ✅ rn ✅ docs
* #19337 ✅ rn ✅ docs
* #19354 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-429.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+* Allow using the `#` and `?` characters in S3 locations. ({issue}`19296`)
+
+## Hive connector
+
+* Add support for `varchar` to `date` coercion. ({issue}`19201`)
+* Add support for `decimal` to `tinyint`, `smallint`, `integer`, and `bigint`
+  coercions for partitioned tables. ({issue}`19201`)
+* Improve performance of reading ORC files. ({issue}`19295`)
+* Allow using the `#` and `?` characters in S3 locations. ({issue}`19296`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-429.md:27]: See above

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-429.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Release 429 (11 Oct 2023)
+
+## General
+
+* Allow {doc}`/sql/show-functions` for a specific schema. ({issue}`19243`)
```
[the remark this answers is not in the document]
Comment (colebow) [reply docs/src/main/sphinx/release/release-429.md:5]: @electrum did write this as schema in his proposed RN.

[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-429.md:5]: I know .. and the docs say the same .. so its probably alright for this .. but I have the larger question for the SQL routine stuff .. worst case I have to update this later

File: docs/src/main/sphinx/release/release-429.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Delta Lake connector
+
+* Allow using the `#` and `?` characters in S3 location paths or URLs. ({issue}`19296`)
+
+## Hive connector
+
+* Add support for `varchar` to `date` coercion. ({issue}`19201`)
+* Add support for `decimal` to `tinyint`, `smallint`, `integer`, and `bigint`
+  coercions for partitioned tables. ({issue}`19201`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-429.md:25]: "Coercions" is the wrong term here. Is this about supporting partitions where the declared type of a column in the partition doesn't match the type in the table? Or for evolving the schema of a table? 

It seems similar to the entry in release 427: "Allow changing a column’s type from double to varchar in Hive tables."

File: docs/src/main/sphinx/release/release-429.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* Fix error reading Avro files when a schema has uppercase characters in its
+  name. ({issue}`19249`)
+
+## Hudi connector
+
+* Allow using the `#` and `?` characters in S3 location paths or URLs. ({issue}`19296`)
+
+## Iceberg connector
+
+* Add support for automatic type coercion when creating Iceberg tables. ({issue}`13981`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-429.md:37]: What does this mean?

Comment (colebow) [reply docs/src/main/sphinx/release/release-429.md]: Completely reworded to be more specific

**Review by mosabua — APPROVED** [review]
Looks good to me now.

File: docs/src/main/sphinx/release/release-429.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+* Log out from a Trino OAuth session when logging out from the Web UI. ({issue}`13060`)
+
+## Delta Lake connector
+
+* Allow using the `#` and `?` characters in S3 location paths or URLs. ({issue}`19296`)
+
+## Hive connector
+
+* Add support for `varchar` to `date` coercion. ({issue}`19201`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-429.md:23]: This one needs to be updated, too.

File: docs/src/main/sphinx/release/release-429.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+* Add changing a column's type from `decimal` to `tinyint`, `smallint`,
```
Comment (martint) [inline docs/src/main/sphinx/release/release-429.md:24]: "Add support for"?

**Pull request merged**